FEAT Adding Python example for unit test with mock (pattern 05)

wtromano-aws opened the pull request:
*Issue #, if available:*

Issue 47

*Description of changes:*

Adds an example of local testing of Lambda using mocks for test pattern 05.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

dancfox commented on python-test-samples/lambda-mock/README.md:
```diff
@@ -0,0 +1,141 @@
+
+The SUT in this pattern is a Lambda function that makes calls to other AWS cloud services using an AWS SDK.
+
+![System Under Test](/_img/pattern_04_lambda_mock_sut.png)
```
Incorrect diagram? This is the same diagram as below. Tests and synthetic event are not part of the SUT.

wtromano-aws replied:
I triple-checked, and this is the diagram from our inventory.  Perhaps the inventory is also incorrect?

dancfox replied:
Okay you are right. The inventory doc is wrong. However, this image does not accurately represent the architecture of the system under test in your sample. Could you create a new image from the PPT that does? 

Representations of tests should not appear in SUT diagrams.

dancfox commented on python-test-samples/lambda-mock/README.md:
````diff
@@ -0,0 +1,141 @@
+
+## Running the project
+
+* First, create the Python Virtual Environment by using the ```make createEnv``` command.
````
wrong make command?

wtromano-aws replied:
I missed a whole section of the README when I re-tooled the Makefile, thank you for finding it.   Resolved.

dancfox commented on python-test-samples/lambda-mock/README.md:
````diff
@@ -0,0 +1,141 @@
+## Running the project
+
+* First, create the Python Virtual Environment by using the ```make createEnv``` command.
+* Run the unit tests with the ```make unittest``` command.
````
wrong command

dancfox commented on python-test-samples/lambda-mock/README.md:
````diff
@@ -0,0 +1,141 @@
+
+* First, create the Python Virtual Environment by using the ```make createEnv``` command.
+* Run the unit tests with the ```make unittest``` command.
+* To deploy the project, deploy the project using the ```make deploy.guided``` command.
````
wrong command

wtromano-aws replied:
same.

dancfox commented on python-test-samples/lambda-mock/README.md:
```diff
@@ -0,0 +1,141 @@
+
+---
+
+## License
```
License not necessary here.

dancfox reviewed (changes requested):
Please see comments! I will do another round once you attend to a few fixes. Thank you!

wtromano-aws commented on python-test-samples/lambda-mock/LICENSE:
```diff
@@ -0,0 +1,15 @@
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Removed, thank you!

wtromano-aws commented on python-test-samples/lambda-mock/Makefile:
```diff
@@ -0,0 +1,74 @@
+
+	Commands:
+
+	build     Build Lambda function dependencies
```
Corrected, great catch.

wtromano-aws commented on python-test-samples/lambda-mock/Makefile:
```diff
@@ -0,0 +1,74 @@
+	deps      Install project dependencies locally
+	test      Run unit test locally using mockets
+	coverage  Run unit test coverage reports
+	scan	  Run cose scanning tools
```
corrected, thank you.

wtromano-aws commented on python-test-samples/lambda-mock/CONTRIBUTING.md:
```diff
@@ -0,0 +1,59 @@
+# Contributing Guidelines
```
removed.

wtromano-aws replied:
removed

wtromano-aws commented on python-test-samples/lambda-mock/Makefile:
```diff
@@ -0,0 +1,74 @@
+	deploy.g  Deploy guided (for initial deployment)
+	deploy    Deploy subsequent changes
+
+	install   Install Pipenv, application and dev dependencies defined in Pipfile
```
copy-paste issue!  resolved in the latest push.

wtromano-aws commented on python-test-samples/lambda-mock/Makefile:
```diff
@@ -0,0 +1,74 @@
+	install   Install Pipenv, application and dev dependencies defined in Pipfile
+	shell     Spawn a virtual environment shell
+	deps      Install project dependencies locally
+	test      Run unit test locally using mockets
```
mockets = mocked muppets?
Great catch, thank you.  Resolved.

dancfox commented:
Ran into the following error.

![Screenshot 2023-03-03 at 2 50 50 PM](https://user-images.githubusercontent.com/2635813/222836387-6adeeed6-b253-44e6-922b-29652405a343.png)

wtromano-aws commented:
Thank you Dan, I updated the diagrams in the inventory and this repo.
Also, I think I addressed the issue in the Makefile, let me know how the edit works for you.
